llvm/lib/Target/X86/X86FastISel.cpp

```diff
@@ -2725,6 +2734,7 @@ bool X86FastISel::fastLowerIntrinsicCall(const IntrinsicInst *II) {
       // without a call if possible.
       uint64_t Len = cast<ConstantInt>(MCI->getLength())->getZExtValue();
       if (IsMemcpySmall(Len)) {
+        annotateIntrinsic(II, "1");
```

Comment:
What do `1` and `0` mean? I am very allergic to these sort of constants in general -- they should probably be `#define`s.

Reply:
"1" means the intrinsic was inlined. They should probably be defined.

Reply:
I realised what it means :) But *why* the string `1` and the string `0` to represent a boolean?

Reply:
I see.

@ptersilie it appears that [metadata nodes can have arbitrary type](https://llvm.org/docs/LangRef.html#metadata), so we could use an `i1`? (there is no such thing as a bool in LLVM IR AFAIK).

Reply:
Fixed in e9b3a75 and https://github.com/ykjit/yk/pull/433/commits/a71d064b6ca6e2f72e424410aed7a0f6615533a9.